compiler/modules/types/function.bal

```diff
@@ -221,6 +221,10 @@ function bddFunctionAtomicType(Env env, Bdd bdd) returns FunctionAtomicType? {
     return ();
 }
 
+function functionSubtypeComplement(ProperSubtypeData t) returns SubtypeData {
+    return bddSubtypeDiff(FUNCTION_TOP, t);
```

Comment:
@jclark I don't think this actually makes a difference since I can replace the `FUNCTION_TOP` with any arbitrary BDD node (say `LIST_SUBTYPE_MAPPING_RO` -> `readonly & (map<readonly>)[]`) and everything works. This is true for table type as well. The only test that fails is [this serialization test](https://github.com/ballerina-platform/nballerina/blob/0331ffaf618ed6e5ece4efa405e8712105cdeabc/compiler/modules/types.sexpr/README.md?plain=1#L143) since they produce slightly different BDDs.